homeassistant/components/shelly/config_flow.py

```diff
@@ -213,7 +214,7 @@ async def async_step_credentials(
                 )
             except InvalidAuthError:
                 errors["base"] = "invalid_auth"
-            except DeviceConnectionError:
+            except (DeviceConnectionError, MacAddressMismatchError):
                 errors["base"] = "cannot_connect"
```

Comment:
Should we make a new error for `mac_address_mismatch` so they don't get confused about why they can connect but its not accepting the device?

Reply:
I thought this should be very rare, but rethinking about it it would be good to create a different error mainly due to it being very rare since it will be hard to get logs with this error and it is better that the user will report the correct error so we can point to the right solution. I will update the PR.